src/tokens/tokens.service.ts

```diff
@@ -432,7 +444,8 @@ export class TokensService {
   }
 
   async supportsData(address: string, type: TokenType) {
-    const iid = type === TokenType.FUNGIBLE ? ERC20WithDataIID : ERC721WithDataIID;
+    const withDataIID = await this.supportsNFTUri(address, false) ? ERC721WithDataUriIID : ERC20WithDataIID
+    const iid = type === TokenType.FUNGIBLE ? ERC20WithDataIID : withDataIID;
```

Comment:
This logic might be correct but seems a little hard to follow... maybe a `switch` would be clearer?

Reply:
tried to make it clearer with a `switch` and better naming